### ISSUE TITLE
JSONConvertion.save should return model instance, not JSON

### ISSUE THREAD

**Issue by ehannes** [my-codeworks/fortnox-api#64]
When saving a new instance of a Fortnox resource, the repository currently returns JSON instead of the saved model.

```
puts API::Fortnox::Repository::Order.new.save(model)
# => {"Order"=>{"@url"=>"https://api.fortnox.se/3/orders/18", "@urlTaxReductionList"=>"https://api.fortnox.se/3/taxreductions?filter=orders&referencenumber=18", "AdministrationFee"=>0, "AdministrationFeeVAT"=>0, "Address1"=>"", "Address2"=>"", "BasisTaxReduction"=>0, "Cancelled"=>false, "City"=>"", "Comments"=>"A great comment about something", ...}}
```

Maybe it should return the saved instance instead?

**Comment by d-Pixie** [comment]
Should return a __new__ instance of the model if the model is new and the __same__ instance that you sent in if the model is just unsaved.

**Comment by ehannes** [comment]
This issue is dependent on #62.

**Comment by ehannes** [comment]
It turns out that `Repository::JSONConvertion.convert_hash_keys_from_json_format( hash, key_map )` does not handle nested models. This should be fairly easy to fix with recursion, but I bumped into yet another problem: `@options.json_to_attr_map`.

`@options.json_to_attr_map` is the inverse of `@options.attr_to_json_map`. It works well for normal attributes, but with nested it doesn't. This is `Order`'s `attr_to_json_map` (I removed some items to make it shorter):
```
{"AdministrationFeeVAT"=>:administration_fee_vat, ... {:vat=>"VAT"}=>:order_rows}
```

Note that the nested `order_rows` becomes weird. What we want it something like `order_rows: { 'VAT' => :vat }`.

**Comment by ehannes** [comment]
@d-Pixie: the question is what `repository.save(old_model)` (that is, updating an existing model) should return. If the returning object is same as we send, some attributes will be incorrect. An example:

Lets create a new `Order`:
```
order_from_fortnox = order_repository.save({ order: { customer_number: 1, order_rows: [ { price: 3, ordered_quantity: 7, delivered_quantity: 7 } ] } } )

puts order_from_fortnox.inspect
# => #<Fortnox::API::Model::Order:0x007fb6c2a7ca08 @customer_number="1", ..., @order_rows=[#<Fortnox::API::Model::OrderRow:0x007fb6c38ee6e0 ..., @ordered_quantity=3.0, @price=7.0, @total=21.0>]>
```

Then I want to update the model:
```
updated_model = order_from_fortnox.update(order_row: { price: 5 } )
puts updated_model.order_row[0].total
# => 21.0
```
Now the `total` for `OrderRow` is incorrect. We want Fortnox to calculate this for us. But if we, as discussed before, return same object here (and not the resource Fortnox is returning), `total` will not be updated, right?
```
order_from_fortnox = order_repository.save(updated_model) # returns same instance, that is order_from_fortnox == updated_model
puts order_from_fortnox.order_rows[0].total
# => 21.0
```

Aren't we forced to always return the instance Fortnox is returning?

**Comment by d-Pixie** [comment]
@ehannes Isn't this one fixed with #82 ? I seem to remember the savers returning instantiated models now :)

**Comment by ehannes** [comment]
@d-Pixie look at the pull request reference above ;)
